alibi/explainers/similarity/grad.py

```diff
@@ -128,6 +129,14 @@ def __init__(self,
             task_name=task
         )
 
+        non_trainable_layers = self.backend.get_non_trainable(self.predictor)
+        if non_trainable_layers:
+            layers_msg = 'The following layers are not trainable: '
+            layers = ", ".join([f"'{layer}'" for layer in non_trainable_layers if layer is not None])
```

Comment:
`if layer is not None` -> are there instances where the returned list containe `None` elements?

Reply:
I haven't been able to figure out this behaviour. `EagerTensors` don't have name attributes, so those layer names returned would be `None`. I'm leaving this in defensively so that if this behaviour does happen then an error won't occur but the user still receives an error message.

I've also added an error message for the case where no parameter in the model is trainable.